webapp/src/main/java/com/box/l10n/mojito/service/asset/AssetService.java

```diff
@@ -396,4 +398,12 @@ public void deleteAssetOfBranch(Long assetId, Long branchId) {
         List<Asset> all = assetRepository.findAll(assetSpecifications);
         return all;
     }
+
+    public Set<Long> findAllAssetIds(Long repositoryId, String path, Boolean deleted, Boolean virtual, Long branchId) {
+        Set<Long> all = new HashSet<>();
+        for (Asset asset: findAll(repositoryId, path, deleted, virtual, branchId)) {
```

Comment:
we just migrated to java 8, if you want you can change it to `findAll(repositoryId, path, deleted, virtual, branchId).stream().map(Asset::getId).collect(Collectors.toSet());`

Reply:
oh, nice.